### PR TITLE
Remove notice on #12 and add a WHY section

### PR DESCRIPTION
**Pull request by fatso83**
Removed the notice on needing more maintainers. 

Added a WHY section so that people understand why they should be using this.

Not super sure on the order of the sections ... feel free to suggest modifications.

### CODE REVIEW

**Comment by RobertWHurst** [comment]
Looks good 👍

File: README.md [hunk 1]
```diff
@@ -1,8 +1,55 @@
...
+Instead of writing tedious setup and teardown code for each
+individual test case you can let Sinon do all the cleanup for you.
+
+So instead of doing this (using [Mocha](https://mochajs.org/) syntax):
+
+```javascript
+var spy1;
+var spy2;
+
+afterEach(()={
```
Comment (RobertWHurst) [inline README.md]: A crap. I missed this. Slight typo ;)

File: README.md [hunk 2]
```diff
@@ -1,8 +1,55 @@
...
+    myFunc(2);
+    assert(spy1.calledWith(1));
+    assert(spy1.calledWith(2));
+});
+```
+
+You could write just this
+
+```javascript
+it('should do something', sinon.test(function(){
```
Comment (RobertWHurst) [inline README.md]: I'd recommend using an arrow function here too.

File: README.md [hunk 3]
```diff
@@ -24,15 +71,15 @@ var assert = require('assert');
 sinon.test = sinonTest.configureTest(sinon);
 sinon.testCase = sinonTest.configureTestCase(sinon);
 
-describe('my function', function() { 
+describe('my function', function() {
```
Comment (RobertWHurst) [inline README.md]: And here.

File: README.md (hunk 2 above)
Comment (fatso83) [inline README.md]: AFAIK that does not work?

**Comment by fatso83** [comment]
It was already merged by the time you commented ...

File: README.md (hunk 2 above)
Comment (RobertWHurst) [inline README.md]: It should work. As far as I know lambdas have a length property and an apply method (even though the context is ignored).

**Pull request merged**